core/thread/test/testRWLock.cxx

```diff
@@ -0,0 +1,398 @@
+// #include "TReentrantRWLock.h"
+#include "TVirtualMutex.h"
+#include "TMutex.h"
+#include "TVirtualRWMutex.h"
+#include "ROOT/TReentrantRWLock.hxx"
+#include "ROOT/TRWSpinLock.hxx"
+
+#include "../src/TRWMutexImp.h"
```

Comment:
It seems that we are missing a concept somewhere in the public implementation. Could you add a comment why we need to do this?